src/cpp/rtps/builtin/discovery/participant/PDPServer.cpp

```diff
@@ -737,16 +749,18 @@ void PDPServer::announceParticipantState(
         std::vector<GUID_t> remote_readers;
         LocatorList locators;
 
-        std::vector<GuidPrefix_t> direct_clients_and_servers = discovery_db_.direct_clients_and_servers();
-        for (GuidPrefix_t participant_prefix: direct_clients_and_servers)
         {
-            // Add remote reader
-            GUID_t remote_guid(participant_prefix, c_EntityId_SPDPReader);
-            remote_readers.push_back(remote_guid);
+            std::vector<GuidPrefix_t> direct_clients_and_servers = discovery_db_.direct_clients_and_servers();
+            for (GuidPrefix_t participant_prefix: direct_clients_and_servers)
+            {
+                // Add remote reader
+                GUID_t remote_guid(participant_prefix, c_EntityId_SPDPReader);
+                remote_readers.push_back(remote_guid);
 
-            locators.push_back(discovery_db_.participant_metatraffic_locators(participant_prefix));
+                locators.push_back(discovery_db_.participant_metatraffic_locators(participant_prefix));
+            }
+            send_announcement(change, remote_readers, locators, dispose);
         }
```

Comment:
What is the purpose of this scope block?

Reply:
Looks like stale code from a previous implementation. Removed.